pipx/commands.py

```diff
@@ -327,6 +335,71 @@ def install(
         venv.remove_venv()
         raise
 
+    # if all is well, write out pipxrc file
+    package_or_url_pipxrc = _abs_path_if_local(package_or_url, venv, pip_args)
+    pipxrc = Pipxrc(venv_dir, read=False)
+    pipxrc.set_package_or_url(package_or_url_pipxrc)
```

Comment:
this feels to my Java-ish, can't we just set them directly? Aka via some properties?

Reply:
I wanted to use a member variable or something like `@property` to disguise getter/setter functions.  The problem was when I wanted to integrate default arguments with the getters.  That broke the simple analog to a member variable.  I wanted to have a default to enable graceful degradation in case a pipxrc file was missing or had missing information.

And then as long as I had a member function for getting, it seemed weird and asymmetric to use something that looked like a member variable just for setting.

Reply:
What kind of default? Where it's set from? I think would be much more readable to have

```python
pipxrc.package_or_url = package_or_url_pipxrc
```
inside the getter what's stopping you from doing:

```
@property
def getter(self):
    if self._package_or_url is None:
       return magic_default
    return self._package_or_url
```

Reply:
It's set from the calling code.  That's why there is no `magic_default`.  If I knew what the magic_default was I wouldn't need to get it as a default argument from the calling code.

Reply:
This was also my attempt to be more explicit for the calling code, I didn't want to be doing something behind its back where it seems like it's getting info from the pipxrc.json file but in reality nothing is there and it's just a default value.

Reply:
In that case why not return None as no value-set and let the caller default to whatever he wants. setter/getter functions feels really non-pythonic to me 🤷‍♂ and itches seeing it

Reply:
Could you do something like `.get('package_or_url', [])` at the callsite? 

If not I'm fine with this. I wouldn't block the PR on it.

Reply:
OK, I'll implement something in the calling code.  I was just trying to be "handy" with something we'd want to do all the time, but it doesn't necessarily need to be part of the pipxrc code.

Reply:
> Could you do something like `.get('package_or_url', [])` at the callsite?

Actually maybe explain this.  Would `get()` be an extra member function of Pipxrc that would do the default substitution for any property with `None`?

Reply:
I did think of one way to do this like a `.get()`.  It would be using `gettattr()` from the calling code.  In Pipxrc I could implement a @property-decorated getter that would throw an `AttributeError` if the internal property was `None`.  That way `getattr()` would substitute its default value.

I don't know if that's trying to be too clever.

Reply:
> Actually maybe explain this. Would get() be an extra member function of Pipxrc that would do the default substitution for any property with None?

That sounds like a decent implementation but honestly I was just thinking of it in terms of the API. Python dict's have that method, and it's Pythonic and quite nice. So I thought if we had it on the pipxrc class it would make the code readable, defaults defined at the callsite, and be Pythonic. 

I see the current code has some if/else statements. I made a comment suggesting it be condensed into one line by using `pipxrc.value or default_value`, which is just sort of a different way of writing `.get()`. 

I would prefer `.get()` but only if it doesn't require too much fancy hacks on the pipxrc class.